fillers/eips/eip4844/blob_txs.py

```diff
@@ -0,0 +1,414 @@
+) -> int:
+    return (
+        tx_max_data_gas_cost
+        if tx_max_data_gas_cost is not None
+        else data_gasprice
+    )
+
+
+@pytest.fixture(autouse=True)
+def txs(  # noqa: D103
+    destination_account: str,
+    tx_count: int,
+    tx_gas: int,
+    tx_value: int,
+    fee_per_gas: int,
+    max_fee_per_gas: Optional[int],
+    max_fee_per_data_gas: int,
+    tx_max_priority_fee_per_gas: int,
+    blobs_per_tx: int,
+    tx_error: str,
+) -> List[Transaction]:
+    if max_fee_per_gas is None:
+        max_fee_per_gas = fee_per_gas
+    return [
+        Transaction(
+            ty=3,
+            nonce=tx_i,
+            to=destination_account,
+            value=tx_value,
+            gas_limit=tx_gas,
+            max_fee_per_gas=max_fee_per_gas,
+            max_priority_fee_per_gas=tx_max_priority_fee_per_gas,
+            max_fee_per_data_gas=max_fee_per_data_gas,
+            access_list=[],
+            blob_versioned_hashes=[
+                to_hash_bytes(x) for x in range(blobs_per_tx)
+            ],
+            error=tx_error if tx_i == (tx_count - 1) else None,
+        )
+        for tx_i in range(tx_count)
+    ]
+
+
+@pytest.fixture
+def account_balance_modifier() -> int:  # noqa: D103
+    return 0
+
+
+@pytest.fixture
+def pre(  # noqa: D103
+    total_account_minimum_balance: int,
+    account_balance_modifier: int,
+) -> Dict:
+    return {
+        TestAddress: Account(
+            balance=total_account_minimum_balance + account_balance_modifier
+        ),
+    }
+
+
+@pytest.fixture
+def env(  # noqa: D103
+    parent_excess_data_gas: Optional[int],
+) -> Environment:
+    return Environment(excess_data_gas=parent_excess_data_gas)
+
+
+@pytest.fixture
+def blocks(  # noqa: D103
+    txs: List[Transaction],
+    tx_error: str,
+) -> List[Block]:
+    return [
+        Block(
+            txs=txs,
+            exception=tx_error,
+        )
+    ]
+
+
+@pytest.mark.parametrize(
+    "parent_excess_blobs,tx_max_data_gas_cost,tx_error",
+    [
+        # tx max_data_gas_cost of the transaction is not enough
+        (
+            12,  # data gas price is 2
+            1,  # tx max_data_gas_cost is 1
+            "insufficient max fee per data gas",
+        ),
+        # tx max_data_gas_cost of the transaction is zero, which is invalid
+        (
+            0,  # data gas price is 1
+            0,  # tx max_data_gas_cost is 0
+            "invalid max fee per data gas",
+        ),
+    ],
+    ids=["insufficient_max_fee_per_data_gas", "invalid_max_fee_per_data_gas"],
+)
+@pytest.mark.parametrize("fork", forks_from(Cancun))
+def test_invalid_tx_max_fee_per_data_gas(
+    blockchain_test: BlockchainTestFiller,
+    pre: Dict,
+    env: Environment,
+    blocks: List[Block],
+    fork: Fork,
+):
+    """
+    Reject blocks with invalid blob txs due to:
+        -
+        - tx max_fee_per_data_gas is not enough
+        - tx max_fee_per_data_gas is zero
+    """
+    blockchain_test(
```

Comment:
If we really wanted to minimize code, we could make a `auto_fill` fixture that does these lines too, but I hesitated to do it.

Reply:
It would be more elegant, but I tend to think it's better to leave it as it is?

I like the thought, but then the test body would simply contain `pass`. This would definitely confuse developers on first view. It's hard for me to judge how confusing it is for the uninitiated to use our `state_test`/`blockchain_test` fixtures now though, tbh :sweat_smile: